src/daft-parquet/src/read.rs

```diff
@@ -155,6 +160,34 @@ async fn read_parquet_single(
     let metadata_num_rows = metadata.num_rows;
     let metadata_num_columns = metadata.schema().fields().len();
 
+    let num_deleted_rows = if let Some(delete_rows) = delete_rows {
```

Comment:
What if `delete_rows` was a bitmap? Then this would be very simple as just a `delete_rows.slice(start_offset, ...)`.

Reply:
I don't think we know the actual number of rows that will be read until it's actually read, so we would need to extend that bitmap. I figured it would be better just to keep it as a vec of indices until we can construct the boolean array

Reply:
I will create a bitmap instead of a vec of bools though